server.js

```diff
@@ -36,37 +76,108 @@ server.post('/api/register', (req, res) => {
         });
 });
 
-server.get('/api/users', (req, res) => {
-    db('users_table').select('id', 'username').then(users => {
-        res.status(201).json(users);
-    }).catch(err => {
-        console.log("error:", err);
-        res.status(500).json(err);
-    })
-});
-
-
 server.post('/api/login', (req, res) => {
+    //grab creds
     const creds = req.body;
 
+    //find the user
     db('users_table')
     .where({username: creds.username})
     .first()
     .then(user => {
+        //check creds
         if (user && bcrypt.compareSync(creds.password, user.password)){
-            res.status(200).send('Welcome');
+            req.session.username = user.username;
+            res.status(200).send(`Welcome ${req.session.username}`);
         } else {
             res.status(401).json({message: 'incorrect combination'});
         }
-    }).catch(err => {res.status(500).send(err, "Everything failed")});
-})
+    }).catch(err => {
+        console.log('/api/login Post error:', err);
+        res.status(500).send(err, "Everything failed")});
+});
+
+server.get('/api/logout', (req, res) => {
+    if (req.session) {
+        req.session.destroy(err => {
+            if (err) {
+                res.send('error logging out');
+            } else {
+                res.send('good bye');
+            }
+        });
+    }
+});
+
+server.get('/api/users', restricted, (req, res) => {
+    
+    if(req.session && req.session.username){
+
+    
+        db('users_table').select('id', 'username', 'password').then(users => {
+            res.status(201).json(users);
+        }).catch(err => {
+            console.log("error:", err);
+            res.status(500).json(err);
+        })
+    
+        }else {
+    
+        res.status(401).json({message: 'Not authorized'});
+        }
+
+    
+});
+
+server.get('/api/admins', restricted, (req, res) => {
+    // grab the logged in userid from the session
+    if(req.session && req.session.userId){
+    const userId = req.session.userId;
+    db('roles as r')
+        .join('user_roles as ur', 'ur.role_id', '=', 'r.id')
+        .select('r.role_name')
+        .where('ur.user_id', userId)
+        .then(roles => {
+            if(roles.includes('admin') || roles.includes('boss')){
+                //have access
+            } else {
+                //bounce
+            }
+        })
+    }
```

Comment:
I think this is a good approach. Another approach is to save the role in `req.session`, like `req.session.role`. Then, you can merely read from the session to continue with authorization.